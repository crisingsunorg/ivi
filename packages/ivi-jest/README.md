# Jest utils for ivi library
