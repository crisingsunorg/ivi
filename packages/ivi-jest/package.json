{
  "private": true,
  "name": "ivi-jest",
  "version": "0.0.1",
  "main": "dist/main/index.js",
  "module": "dist/module/index.js",
  "sideEffects": false,
  "files": [
    "dist",
    "src",
    "!dist/**/*.tsbuildinfo",
    "!src/**/__tests__",
    "README.md",
    "LICENSE"
  ],
  "scripts": {
    "prepublishOnly": "npm run clean && npm run dist",
    "clean": "rm -rf ./dist",
    "dist": "npm run dist:main && npm run dist:module",
    "dist:main": "../../node_modules/.bin/tsc -p ./tsconfig.build.main.json",
    "dist:module": "../../node_modules/.bin/tsc -p ./tsconfig.build.module.json"
  },
  "dependencies": {
    "jsdom": "^14.0.0"
  },
  "peerDependencies": {},
  "devDependencies": {},
  "description": "Jest utils for ivi library",
  "license": "MIT",
  "author": {
    "name": "Boris Kaul",
    "email": "localvoid@gmail.com",
    "url": "https://github.com/localvoid"
  },
  "keywords": [
    "ivi"
  ],
  "homepage": "https://github.com/localvoid/ivi",
  "repository": {
    "type": "git",
    "url": "https://github.com/localvoid/ivi"
  },
  "bugs": {
    "url": "https://github.com/localvoid/ivi/issues"
  },
  "github": "https://github.com/localvoid/ivi"
}
