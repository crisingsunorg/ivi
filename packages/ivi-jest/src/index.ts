export { useMockFn, useSpyOn } from "./mock";
export { useResetModules, useModule } from "./module";
export { useDOMElement, useDOMOpsCounters } from "./dom";
