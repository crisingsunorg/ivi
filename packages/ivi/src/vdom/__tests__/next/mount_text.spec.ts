import { useResetModules, useModule, useDOMElement } from "ivi-jest";

useResetModules();
const c = useDOMElement();
const t = useModule<typeof import("ivi-test")>("ivi-test");

describe("mount text node", () => {
  test("empty string", () => {
    const { domNode } = t.render("", c());
    expect(domNode).toMatchSnapshot();
  });

  test("string", () => {
    const { domNode } = t.render("abc", c());
    expect(domNode).toMatchSnapshot();
  });

  test("zero number", () => {
    const { domNode } = t.render(0, c());
    expect(domNode).toMatchSnapshot();
  });

  test("number", () => {
    const { domNode } = t.render("123", c());
    expect(domNode).toMatchSnapshot();
  });
});
