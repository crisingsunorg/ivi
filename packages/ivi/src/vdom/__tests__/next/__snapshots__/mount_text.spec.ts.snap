// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mount text node empty string 1`] = ``;

exports[`mount text node number 1`] = `123`;

exports[`mount text node string 1`] = `abc`;

exports[`mount text node zero number 1`] = `0`;
